ols/src/prompts/prompt_generator.py

```diff
@@ -34,8 +45,24 @@ def prompt_for_configuration(
     default_prompt: ChatPromptTemplate,
 ) -> ChatPromptTemplate:
     """Find prompt for given configuration parameters or return the default one."""
-    selector = PromptConfiguration(provider, model, rag_exists, history_exists)
-    return prompt_configurations.get(selector, default_prompt)
+    # construct system prompt first
+    selector = PromptConfiguration(provider, model)
+    system_prompt = prompt_configurations.get(selector, default_prompt)
+
+    # add optional parts into system prompt
+    if history_exists:
+        system_prompt += USE_PREVIOUS_HISTORY
+    if rag_exists:
+        system_prompt += USE_RETRIEVED_CONTEXT
+
+    # construct chat prompt from sequence of messages
+    messages = []
+    messages.append(SystemMessagePromptTemplate.from_template(system_prompt))
+    if history_exists:
+        messages.append(MessagesPlaceholder(variable_name="chat_history"))
+    messages.append(HumanMessagePromptTemplate.from_template("{query}"))
+
+    return ChatPromptTemplate.from_messages(messages)
 
 
 def generate_prompt(
```

Comment:
a number of the args to generate_prompt do not appear to be used, e.g. `query`, `conversation_id`, `referenced_documents`, and `model_options`

is there some future plan for their use or am i missing something?

Reply:
yes, we talked about this with Asutosh, these are there for usage in the future.

Reply:
i don't love it, but if that usage is coming within the next sprint i guess it's ok.  If it's potentially further out than that, please remove them until we have an actual need for them.

Reply:
removed unused params we know won't be used soon